File: Backend/Farmer_Service/src/main/java/com/example/demo/entity/ProductSeller.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "product_seller")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class ProductSeller {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int psId;

    private int price;
    private int qty;

    @ManyToOne
    @JoinColumn(name = "p_id", referencedColumnName = "pid") // FK to product table
    private Product product;

    @ManyToOne
    @JoinColumn(name = "u_id", referencedColumnName = "uid") // FK to farmer/user table
    private Farmer farmer;

    @Column(name = "available")
    private boolean available = true;
    // Getters and setters
}

File: Admin_Backend/Models/ProductSeller.cs
using System;
using System.Collections.Generic;

namespace Admin_Backend.Models
{
    public partial class ProductSeller
    {
        public ProductSeller()
        {
            OrderDetails = new HashSet<OrderDetail>();
        }

        public int PsId { get; set; }
        public double Price { get; set; }
        public int Qty { get; set; }
        public int? PId { get; set; }
        public int? UId { get; set; }
        public ulong? Available { get; set; }

        public virtual Product? PIdNavigation { get; set; }
        public virtual User? UIdNavigation { get; set; }
        public virtual ICollection<OrderDetail> OrderDetails { get; set; }
    }
}
